### PR TITLE
feat: Filter RFQM quotes with the wrong chain id

### PR DESCRIPTION
**Pull request by rhinodavid**
# Description

Closes MKR-78.

When a MM posts a quote, compare the chain id of the quote with the environment chain id. If they are not the same, do not include the quote in the best quotes and log a message.

# Testing Instructions

Run new unit test.

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/services/rfqm_service.ts [hunk 1]
```diff
@@ -270,9 +271,18 @@ export class RfqmService {
             opts,
         );
 
+        const firmQuotesWithCorrectChainId = firmQuotes.reduce((result, quote) => {
+            if (quote.order.chainId !== CHAIN_ID) {
+                logger.debug(`Received a quote with incorrect chain id: ${quote}`);
```
Comment (rhinodavid) [inline src/services/rfqm_service.ts:276]: Just want to make sure we want this at the `debug` level

File: test/rfqm_test.ts [hunk 2]
```diff
@@ -551,7 +552,7 @@ describe(SUITE_NAME, () => {
                 takerToken: contractAddresses.etherToken,
                 pool: '0x',
                 salt: '0',
-                chainId: 1,
+                chainId: 1337,
```
Comment (rhinodavid) [inline test/rfqm_test.ts:555]: This is the chain id specified in the test config

File: src/services/rfqm_service.ts (hunk 1 above)
Comment (PirosB3) [reply src/services/rfqm_service.ts:276]: this should always be a scary `error` imo. It should never happen, and if it does we (or a market maker) should adjust immediately.

File: src/services/rfqm_service.ts [hunk 3]
```diff
@@ -270,9 +271,18 @@ export class RfqmService {
             opts,
         );
 
+        const firmQuotesWithCorrectChainId = firmQuotes.reduce((result, quote) => {
```
Comment (PirosB3) [inline src/services/rfqm_service.ts:274]: Why not use `firmQuotes.filter` ?

File: test/rfqm_test.ts [hunk 4]
```diff
@@ -641,6 +642,108 @@ describe(SUITE_NAME, () => {
...
+                        },
+                        responseCode: 200,
+                        responseData: {
+                            signedOrder: {
+                                ...BASE_SIGNED_ORDER,
+                                makerAmount: correctChainQuote,
+                                takerAmount: sellAmount,
+                                signature: {
+                                    ...VALID_SIGNATURE,
+                                    r: '0xb1',
```
Comment (PirosB3) [inline test/rfqm_test.ts:719]: I think that `r` and `s` needs to be the same length as a legit signature, otherwise it won't pass the JSON schema check

**Review by PirosB3 — APPROVED** [review]

File: test/rfqm_test.ts (hunk 4 above)
Comment (rhinodavid) [reply test/rfqm_test.ts:719]: Hmmm... seems to work for me. it's copied from the test above.

File: src/services/rfqm_service.ts (hunk 3 above)
Comment (rhinodavid) [reply src/services/rfqm_service.ts:274]: good call

File: src/services/rfqm_service.ts (hunk 1 above)
Comment (rhinodavid) [reply src/services/rfqm_service.ts]: done

**Pull request merged**